### PR TITLE
Refactor pip installation logic for supporting uninstall

### PR DESCRIPTION
**Pull request by goooxu**

### CODE REVIEW

File: deployment/docker/Dockerfile.build.base [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Copyright (c) Microsoft Corporation
```
Comment (yds05) [inline deployment/docker/Dockerfile.build.base]: Seems this dockerfile.base is already removed by Shinai?

File: src/nni_manager/training_service/pai/paiData.ts [hunk 2]
```diff
@@ -56,7 +56,7 @@ if python3 -c 'import nni' > /dev/null 2>&1; then
   return
 else
   # Install nni
-  pip3 install -v --user git+https://github.com/Microsoft/nni.git@v0.2
+  pip3 install -v --user nnipkg
```
Comment (QuanluZhang) [inline src/nni_manager/training_service/pai/paiData.ts]: typo?

File: Makefile [hunk 3]
```diff
@@ -13,59 +13,26 @@ _END := $(shell echo -e '\e[0m')
 ## Install directories
 ifeq ($(shell id -u), 0)  # is root
     _ROOT := 1
-    BIN_PATH ?= /usr/bin
-    INSTALL_PREFIX ?= /usr/share
-    EXAMPLES_PATH ?= $(INSTALL_PREFIX)/nni/examples
+    BIN_FOLDER ?= /usr/local/bin
```
Comment (yds05) [inline Makefile]: Why we change bin path for root user?

File: README.md [hunk 4]
```diff
@@ -21,15 +21,28 @@ The tool dispatches and runs trial jobs that generated by tuning algorithms to s
...
+    source install.sh
 ```
 
-**verify install**
+**Verify install**
 * The following example is an experiment built on TensorFlow, make sure you have `TensorFlow installed` before running it.
+* And download the examples via clone the source code
+```bash
+    cd ~
+    git clone https://github.com/Microsoft/nni.git@v0.3
```
Comment (QuanluZhang) [inline README.md]: git clone -b v0.3 ...

File: Makefile [hunk 5]
```diff
@@ -9,63 +9,29 @@ _INFO := $(shell echo -e '\e[1;36m')
...
-    NNI_NODE := node
-    NNI_YARN := yarnpkg
-endif
-
-
-# Setting variables end
-
+$(info $(_INFO) Installing dependencies, use local toolchain $(_END))
+NNI_NODE_TARBALL ?= /tmp/nni-node-linux-x64.tar.xz
+NNI_NODE_FOLDER = /tmp/nni-node-linux-x64
```
Comment (yds05) [inline Makefile]: Shall we define a var like $(OS_PLATFORM) to mark for both Linux and Darwin?

**Pull request merged**